snapcraft/internal/repo/_platform.py

```diff
@@ -30,12 +30,12 @@
 
 def _is_deb_based(distro=None):
     if not distro:
-        distro = get_os_release_info()['ID']
+        distro = OsRelease().id()
     return distro in _DEB_BASED_PLATFORM
 
 
 def _get_repo_for_platform():
-    distro = get_os_release_info()['ID']
+    distro = OsRelease().id()
```

Comment:
How about a `.like()` method that looks at `ID` as well as `ID_LIKE` so we can support derivatives more easily? I essentially suggested the same in #1637 but with the new class we could simplify the calling site by baking it into the API. Thoughts?

Reply:
I think that's an interesting idea for #1637, but doing it here would inflate the API with no users.